File: ruby/JsonBase.rb
# https://github.com/MSFPT/JsonBase

require 'json';

class JsonBase

  def initialize file

    @file = file

    if @file.split('.')[-1]!='json'
      
      puts "[Errno 2] No such file or directory: '"+@file+"'"
    
      exit true

    end
    
    begin
    
      File.read @file;
    
    rescue Errno::ENOENT => e
    
      File.open(@file, 'w') { |file| file.write('') }
    
    end

  end

  def get

    file = File.read @file
    
    begin 
      return JSON.parse file
    rescue JSON::ParserError => e
      return {}
    end

  end

  def clear

    File.open(@file, 'w') { |file| file.write('') }
  
  end

  def commit data

    data = JSON.generate data

    File.open(@file, 'w') { |file| file.write(data) }

    return nil

  end

end

File: php/JsonBase.php
<?php 

  /* https://github.com/MSFPT/JsonBase */

  header('Access-Control-Allow-Origin: *');

  class JsonBase{

    public $file_db;

    public function __construct(string $file){
      
      if (!file_exists($file)) { $log = fopen($file,'w+');fwrite($log, '');fclose($log); }

      $this->file_db = $file;
    
    }

    public function get(){
    
      return json_decode(file_get_contents($this->file_db),true) ;
    
    }
    
    public function commit(Array $data){
    
      $myfile = fopen($this->file_db, "w") or die("Unable to open file!");
    
      fwrite($myfile, json_encode($data));
    
      fclose($myfile);
    
    }

    public function clear(){
    
      $this.commit('') ;
    
    }
  
  }

?>
